src/entry-server.js

```diff
@@ -32,9 +32,14 @@ export default function initializeServer(createApp, serverOpts) {
                     components
                         .filter(c => 'vuex' in c)
                         .forEach((c) => {
-                            opts.logger.info('Registering dynamic Vuex module:', c.vuex.moduleName);
-                            store.registerModule(c.vuex.moduleName, c.vuex.module, {
-                                preserveState: store.state[c.vuex.moduleName] != null,
+                            // Allow a function to be passed that can generate a route-aware
+                            // module name
+                            const moduleName = typeof c.vuex.moduleName === 'function' ?
+                                c.vuex.moduleName(router.currentRoute) :
+                                c.vuex.moduleName;
+                            opts.logger.info('Registering dynamic Vuex module:', moduleName);
+                            store.registerModule(moduleName, c.vuex.module, {
+                                preserveState: store.state[moduleName] != null,
```

Comment:
Server side is easy because there's never any prior modules as it's a fresh render each time :)